core/server/worker/src/main/java/alluxio/worker/block/DefaultBlockWorker.java

```diff
@@ -447,15 +447,19 @@ public boolean openUfsBlock(long sessionId, long blockId, OpenUfsBlockOptions op
 
   @Override
   public void closeUfsBlock(long sessionId, long blockId)
-      throws BlockAlreadyExistsException, InvalidWorkerStateException, IOException,
-      WorkerOutOfSpaceException {
+      throws BlockAlreadyExistsException, IOException, WorkerOutOfSpaceException {
     mUnderFileSystemBlockStore.closeReaderOrWriter(sessionId, blockId);
     if (mBlockStore.getTempBlockMeta(sessionId, blockId) != null) {
       try {
         commitBlock(sessionId, blockId);
       } catch (BlockDoesNotExistException e) {
         // This can only happen if the session is expired. Ignore this exception if that happens.
         LOG.warn("Block {} does not exist while being committed.", blockId);
+      } catch (InvalidWorkerStateException e) {
+        // This can happen if there are multiple sessions writing to the same block.
+        // BlockStore#getTempBlockMeta does not check whether the temp block belongs to
+        // the sessionId.
+        LOG.debug("Invalid worker state while committing block.", e);
```

Comment:
Also, wouldn't this be overkill that some other invalidate worker states can be caught?